core/src/hlist.rs

```diff
@@ -1433,6 +1486,26 @@ where
     }
 }
 
+#[cfg(feature = "typenum")]
```

Comment:
A bit out of scope for this PR, but I feel this should return an array of known-length, and it be up to the library user to say "no, I want it as a Vec".

Perhaps a middle ground though: implement the into array, and into vec is provided as a conveniece wrapper arround `let ??? = list.into(); let vector = ???.into();`